src/torchmetrics/functional/image/d_s.py

```diff
@@ -0,0 +1,263 @@
+# Copyright The Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Dict, Tuple
+
+import torch
+from torch import Tensor
+from typing_extensions import Literal
+
+from torchmetrics.functional.image.uqi import universal_image_quality_index
+from torchmetrics.utilities.distributed import reduce
+from torchmetrics.utilities.imports import _TORCHVISION_AVAILABLE
+
+
+def _spatial_distortion_index_update(preds: Tensor, target: Dict[str, Tensor]) -> Tuple[Tensor, Dict[str, Tensor]]:
+    """Update and returns variables required to compute Spatial Distortion Index.
+
+    Args:
+        preds: High resolution multispectral image.
+        target: A dictionary containing the following keys:
+
+            - ``'ms'``: low resolution multispectral image.
+            - ``'pan'``: high resolution panchromatic image.
+            - ``'pan_lr'``: (optional) low resolution panchromatic image.
```

Comment:
why not have rather ms, pan, pan_lr as arguments?

Reply:
The interface is common to other metrics, including the functional and class interface. Changing the interface affects less on the `src` side, but more on the `tests` side. It requires the helper functions (e.g. `MetricTester.run_functional_metric_test`) to support such variety of arguments

Reply:
I think that for private function we can be more specific if it improves readability